CMakeLists.txt

```diff
@@ -311,7 +314,7 @@ endif()
 # FIXME: Support ARM "-mfpu=vfp -mfloat-abi=hard"
 if (("${TARGET_ARCHITECTURE}" STREQUAL "x86_64") OR ("${TARGET_ARCHITECTURE}" STREQUAL "i686"))
   if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang"))
-    set(SSE_FLAGS "-mfpmath=sse" "-msse" "-msse2")
+    set(SSE_FLAGS "-mfpmath=sse" "-msse" "-msse2" "-std=c++0x")
```

Comment:
@NikolajBjorner Thanks. Is there also a corresponding changes to the python/Makefile build system. I did a quick skim of this very large PR and I couldn't see it. We should keep the build systems in sync.

Requiring a C++11 capable compiler is a breaking change. AFAIK this hasn't been required up until now. I have no problem with this requirement because most modern C++ compilers support C++11.

I do worry if such a change would break existing code. For example will the introduction of C++11's implicit move constructors silently break existing code?